Task 4

maximsan opened the pull request:
lambda for CRUD operations  - [lambda](https://0nw7hvxew9.execute-api.eu-west-1.amazonaws.com/dev/products)
FE pull-request - [link](https://github.com/maximsan/nodejs-aws-fe/pull/2)
Cloudfront - [link](dg2e8nno6k0ty.cloudfront.net)
POST/products lambda functions returns error 400 status code if product data is invalid
All lambdas return error 500 status code on any error (DB connection, any unhandled error in code)
All lambdas do console.log for each incoming requests and their arguments

maximsan committed:
feature: add createProducts.sql, createStocks.sql sql scripts

maximsan committed:
feature: integrate getProductById.js, getProductsList.js with AWS db

maximsan committed:
typo: fix typo in db.js

maximsan committed:
feature: update setup with dotenv, serverless-dotenv plugin

maximsan committed:
feature: adjust handlers

maximsan committed:
feature: add createProduct.js handler

maximsan committed:
feature: add logging in handlers

maximsan committed:
fix: change primary key in createStocks.sql

maximsan committed:
fix: change default status code in errorHandler.js

maximsan committed:
docs: update README.md

maximsan committed:
fix: roll-back cors true in serverless.yml for all handlers

maximsan committed:
feat: add validation

asamurai commented:
<img width="962" alt="Снимок экрана 2020-11-09 в 12 48 05" src="https://user-images.githubusercontent.com/19266751/98532076-de472300-2289-11eb-9aa2-fedb3bf243e9.png">
product list link returns internal server error
please fix it

also provide please link on your CloudFront

maximsan committed:
feature: add product validation schema

IgorByl commented on product-service/src/utils/errorHandler.js:
```diff
@@ -0,0 +1,16 @@
+    if (error instanceof NotFoundError) {
+        return createResponse(404, error.message);
+    }
+    if (error instanceof BadRequest) {
```
checking of the instances' classes goes against solid

maximsan replied:
Thanks for review. SOLID includes 5 statements and your assertion looks a little surface. Which solid statement does it ruin? The same approach we use when checking objects against interfaces. Maybe you have something to propose ?

IgorByl replied:
Hi, the Liskov Substitution Principle restricts class instance type checking.

maximsan commented:
> <img alt="Снимок экрана 2020-11-09 в 12 48 05" width="962" src="https://user-images.githubusercontent.com/19266751/98532076-de472300-2289-11eb-9aa2-fedb3bf243e9.png">
> 
> product list link returns internal server error please fix it
> also provide please link on your CloudFront

It was database related issue. I changed pass. It's working. Cloudfront link is in FE pr, but I've added it as well.
